### PR TITLE
Update logos site-wide with new OpenSearch branding

### LINKED ISSUE #370: Update OpenSearch logos site-wide with new branding
**Is your feature request related to a problem? Please describe.**

OpenSearch Dashboards currently uses placeholder logos (e.g. "heatmap") that were set in the initial forking/renaming project: #37

**Describe the solution you'd like**

New icons/logos have been created for the public OpenSearch branding: https://opensearch.org/brand.html

All of the previous changes using the "heatmap" icon for OpenSearch and OpenSearch Dashboards need to be replaced with the updated logos.

**Describe alternatives you've considered**

N/A

**Additional context**

Commit to change logos for forking/renaming project: https://github.com/opensearch-project/OpenSearch-Dashboards/commit/905ccbe698ce88b3309c6b1f528e9b3a106f5d26

Below are examples of the new marks/logos:

#### Default Mark
opensearch_mark_default

#### OpenSearch Dashboards Logo (Dark Mode)
opensearch_dashboards_logo_darkmode

### PR DESCRIPTION
**Pull request by tmarkley**
### Description
* Overall, I couldn't figure out a way to import a custom SVG as an EUI iconType and include it with CSS. I resorted to importing the SVG files and using them directly in the EUI components as stated in the documentation.
* The header logo component would not accept an SVG so I created a React SVG component instead.
* The elastic logo SVG code was still in the Template component so I replaced that with the OpenSearch mark.
* There is still a reference to the heatmap icon in src/plugins/home/public/application/components/tutorial/introduction.test.js but it's unclear how that is used outside of the test or whether it needs to be changed.
* The welcome logo needed to use the centered mark because of the spacing, but that required changing the padding.
* I don't yet know how to test the icons used for each of the plugins: dev_tools, opensearch_logs, opensearch_metrics, or the management plugin.

Brand Guidelines: https://opensearch.org/brand.html

#### Loading
Screen Shot 2021-05-25 at 3 23 09 PM

#### Welcome
Screen Shot 2021-05-25 at 3 13 49 PM

#### Header
Screen Shot 2021-05-24 at 2 43 54 PM

 
### Issues Resolved
#370 
 
### Check List
- [x] New functionality includes testing.
  - [x]  Unit tests pass
  - [x] Integration tests pass
  - [x] Functional tests pass (WIP)
- [x] New functionality has been documented.
  - [x] New functionality has javadoc added (N/A)
- [x] Commits are signed per the DCO using --signoff

### CODE REVIEW

File: src/core/public/chrome/ui/header/branding/opensearch_dashboards_logo_darkmode.tsx [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+ */
+
+/*
+ * Modifications Copyright OpenSearch Contributors. See
+ * GitHub history for details.
+ */
+import React from 'react';
+
+export function OpenSearchDashboardsLogoDarkMode() {
+  return (
```
Comment (kavilla) [inline src/core/public/chrome/ui/header/branding/opensearch_dashboards_logo_darkmode.tsx:35]: I think for draft purposes it's probably like this but we can probably be consistent with the other svgs were an svg file was created and accessed in the react component (without the helper functions):

```
addBasePath(
 `/plugins/${PLUGIN_ID}/assets/opensearch_dashboards_${appId}_${
   IS_DARK_THEME ? 'dark' : 'light'
   }.svg`
 )}
  ```
  
  then we just make this opensearch_dashboards_logo.tsx

**Review by kavilla — COMMENTED** [review]
Functional tests are passing. I think overall looks good. It mimics how svgs are import EUI but can probably have some conditionality based on if the user is using darkmode or not

File: src/core/public/chrome/ui/header/branding/opensearch_dashboards_logo_darkmode.tsx (hunk 1 above)
Comment (tmarkley) [reply src/core/public/chrome/ui/header/branding/opensearch_dashboards_logo_darkmode.tsx:35]: Are you saying that this code would go in `header_logo.tsx`? Where do the `PLUGIN_ID`, `appId`, and `IS_DARK_THEME` props get passed in to the component from?

Comment (tmarkley) [reply src/core/public/chrome/ui/header/branding/opensearch_dashboards_logo_darkmode.tsx:35]: And how do I test enabling/disabling dark mode?

Comment (tmarkley) [reply src/core/public/chrome/ui/header/branding/opensearch_dashboards_logo_darkmode.tsx:35]: In order to get these changes out for 1.0, I've added a follow-up task to the meta issue to address this feature for darkmode/default logos based on the theme: https://github.com/opensearch-project/OpenSearch-Dashboards/issues/372

**Review by ananzh — APPROVED** [review]
LGTM

**Review by kavilla — APPROVED** [review]
Followup and improvements can be made but good first pass for replacing the heatmap.

**Pull request merged**